docs/catalog/cloud/evaluate-using-a-service-mesh.md

```diff
@@ -0,0 +1,42 @@
+---
+version: 1.0
+submitted_by: yelghali
+published_date: tbd
+category: cloud
+tags: 
+- cloud
+- security
+- network
+- kubernetes
+- role:cloud-engineer
+- size:medium
+---
+
+# Use a service mesh only if needed
+
+## Description
+A service mesh deploys additional containers for communication, typically in a sidecar pattern, to provide more operational capabilities leading to an increase in CPU usage and network traffic. 
+
+Nevertheless, it allows you to decouple your application from these capabilities as it moves them out from the application layer, and down to the infrastructure layer.
+
+## Solution
+- Use a service mesh if your application lacks features that can be added by one, such as distributed tracing or traffic encryption.
+- A service mesh can typically be enabled only for some resources, which allows you to use it for required applications only, and not the entire platform.
```

Comment:
Some things can also be done inside the application itself, without using a service mesh. For example, other service DNS names could be injected via configuration
For example:
* a ConfigMap -> Environment Variable in Kubernetes or
* A custom configured DNS Server that answers with the correct IPs (DNS servers are mostly needed in private/corporate environments)

Reply:
How would you add this here in this solution section?

Reply:
Putting this under considerations is fine for me.